packages/flutter/test/widgets/raw_menu_anchor_test.dart

```diff
@@ -0,0 +1,5787 @@
+    expect(find.byType(Overlay), findsNWidgets(2));
+
+    final Iterable<Overlay> overlays = tester.widgetList<Overlay>(find.byType(Overlay));
+    final Overlay nonRootOverlay = tester.widget(find.byKey(overlayKey));
+    final Overlay rootOverlay = overlays.firstWhere((Overlay overlay) => overlay != nonRootOverlay);
+
+    // Check that the ancestor _RenderTheater for the menu item is the one
+    // from the root overlay.
+    expect(
+      ancestorRenderTheaters(tester.renderObject(find.byKey(menuItemKey))).single,
+      tester.renderObject(find.byWidget(rootOverlay)),
+    );
+  });
+
+  group('[Default] Layout', () {
+    final List<AlignmentGeometry> alignments = <AlignmentGeometry>[
+      for (double x = -2; x <= 2; x += 1)
+        for (double y = -2; y <= 2; y += 1)
+          Alignment(x, y),
+      for (double x = -2; x <= 2; x += 1)
+        for (double y = -2; y <= 2; y += 1)
+          AlignmentDirectional(x, y),
+    ];
+
+    /// Returns the rects of the menu's contents. If [clipped] is true, the
+    /// rect is taken after UnconstrainedBox clips its contents.
+    List<Rect> collectOverlays({bool clipped = true}) {
+      final List<Rect> menuRects = <Rect>[];
+      final Finder finder = clipped ? findMenuPanel() : findOverlayContents();
+      for (final Element candidate in finder.evaluate().toList()) {
+        final RenderBox box = candidate.renderObject! as RenderBox;
+        final Offset topLeft = box.localToGlobal(box.size.topLeft(Offset.zero));
+        menuRects.add(topLeft & box.size);
+      }
+      return menuRects;
+    }
+
+    testWidgets('LTR alignment', (WidgetTester tester) async {
+      Widget buildApp({AlignmentGeometry? alignment}) {
+        return App(
+          RawMenuAnchor(
+            alignment: alignment,
+            menuAlignment: Alignment.center,
+            surfaceDecoration: const BoxDecoration(),
+            menuChildren: <Widget>[
+              Container(
+                width: 100,
+                height: 50,
+                color: const Color(0xFF0000FF),
+                child: Text(Tag.a.text),
+              )
+            ],
+            child: const AnchorButton(Tag.anchor),
+          ),
+        );
+      }
+
+      await tester.pumpWidget(buildApp());
+      await tester.tap(find.text(Tag.anchor.text));
+      await tester.pump();
+
+      // Anchor position is fixed.
+      final ui.Rect anchorRect = tester.getRect(
+        find.widgetWithText(Button, Tag.anchor.text),
+      );
+
+      for (final AlignmentGeometry alignment in alignments) {
+        await tester.pumpWidget(buildApp(alignment: alignment));
+        final ui.Rect overlay = tester.getRect(
+          find.widgetWithText(Container, Tag.a.text).first,
+        );
+        expect(
+          alignment.resolve(TextDirection.ltr).withinRect(anchorRect),
+          overlay.center,
+          reason: 'Anchor alignment: $alignment \n'
+              'Menu rect: $overlay \n',
```

Comment:
Done! By the way, thank you so much for taking the time to review everything. I realize it's a gigantic PR, which is never pleasant.

Reply:
Sure it is gigantic, but in that particular case it would be very hard to split it (for instance this raw_anchor_test can not be split on several PRs, maybe the examples could have been in a different PR but this is really not a big deal).

What is pleasant to me is seeing such a great effort making progress. Reviewing would be a pain if we had to comment on many places, but here the quality is very high so it is mainly double-checking and finding places to discuss.
I will try to dedicate time in the coming days to make progress on the review.
Thanks for your patience.